client/pages/results.jsx

```diff
@@ -75,23 +76,47 @@ export default class Results extends React.Component {
     this.setState({ input: event.target.value });
   }
 
+  handleClick(event) {
+    event.preventDefault();
+    const timeStamp = new Date().toLocaleString();
+    const postData = {
+      userId: 1,
+      walletAddress: this.state.address,
+      data: this.state.walletData,
+      bookmarkedAt: timeStamp
+    };
+    fetch('/api/bookmarks', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(postData)
+    })
+      .then(this.setState({ star: true }));
```

Comment:
This is setting state before your fetch completes. `.then()` takes a function _definition_ but here you are passing the return value of a function _call_.